### PR TITLE
Enable object-short-hand linting rule

### PR DESCRIPTION
**Pull request by claire256**
issue - https://github.com/WikiEducationFoundation/WikiEduDashboard/issues/1408#issue-259228085

## What this PR does
This PR is enabling the  Object-Short-hand JavaScript linting rule. 
Object-Short-hand require or disallow method and property shorthand syntax for object literals. In this PR i'm only enforcing the use of shorthand syntax on only 'properties' defined where the key name matches name of the assigned variable. 

The default rule (always), would apply to all 'methods' (including generators) defined in object literals and any 'properties'

PROPERTIES
Example ;
    
 *before:
              var foo = {
                    x: x,
                    y: y,
                    z: z,
                };
 
 *after
             var foo = {x, y, z};    

METHODS       
Example ;
    
 *before:
              var foo = {
                    a: function() {},
                    b: function() {}
                    };

 *after
     var foo = {
              a() {},
              b() {}
             };          
      
#Why
I did not enable the default rule (leaving out the methods), because i felt it would bring violations that may be confusing to some people, so i felt it was not necessary.

### CODE REVIEW

**Comment by claire256** [comment]
What do you think about this linting rule? @ragesoss

File: .eslintrc [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "import/no-named-as-default": 0,
     "no-underscore-dangle": 0,
     "no-empty-pattern": 0,
-    "object-shorthand": 0,
+    "object-shorthand": 2,
```
Comment (ragesoss) [inline .eslintrc:49]: Is `2` the default? We want to enable rules by removing their entries from the dot file, rather than hard-coding their settings.

**Comment by ragesoss** [comment]
I'm fine with enabling this rule.

File: .eslintrc (hunk 1 above)
Comment (claire256) [reply .eslintrc]: 'Always' is the default, but i have changed it to only 'properties'

**Comment by ragesoss** [comment]
> I did not enable the default rule because this would bring complicated violations which could also result in changing the purpose of the code.

Can you explain more? I don't know what the default rule is or what kinds of violations it would trigger.

**Comment by claire256** [comment]
I have explained more. @ragesoss

**Comment by ragesoss** [comment]
Thanks. How many violations of the default rule would there be?

**Comment by claire256** [comment]
They're (93) for default and (90) for properties only. @ragesoss

**Comment by ragesoss** [comment]
@claire256 so there are only three cases where the methods would need to be changed to fit the rule? If that's the case, I'd like to see what it would look like to make those changes.

**Comment by claire256** [comment]
Hello @ragesoss, I have enabled all the default changes.

File: .eslintrc [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "import/no-named-as-default": 0,
     "no-underscore-dangle": 0,
     "no-empty-pattern": 0,
-    "object-shorthand": 0,
+    "object-shorthand": ["error", "always"],
```
[the remark this answers is not in the document]
Comment (claire256) [reply .eslintrc:49]: Alright.

**Pull request merged**